homeassistant/components/bang_olufsen/event.py

```diff
@@ -0,0 +1,88 @@
+"""Event entities for the Bang & Olufsen integration."""
+
+from __future__ import annotations
+
+from homeassistant.components.event import EventDeviceClass, EventEntity
+from homeassistant.const import CONF_MODEL
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BangOlufsenConfigEntry
+from .const import (
+    CONNECTION_STATUS,
+    DEVICE_BUTTON_EVENTS,
+    DEVICE_BUTTONS,
+    MODEL_SUPPORT_DEVICE_BUTTONS,
+    MODEL_SUPPORT_MAP,
+    WebsocketNotification,
+)
+from .entity import BangOlufsenEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: BangOlufsenConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Sensor entities from config entry."""
+
+    entities: list[EventEntity] = []
+
+    # Add physical "buttons"
+    if config_entry.data[CONF_MODEL] in MODEL_SUPPORT_MAP[MODEL_SUPPORT_DEVICE_BUTTONS]:
+        entities.extend(
+            [
+                BangOlufsenButtonEvent(config_entry, button_type)
+                for button_type in DEVICE_BUTTONS
+            ]
+        )
+
+    async_add_entities(new_entities=entities)
+
+
+class BangOlufsenEvent(BangOlufsenEntity, EventEntity):
+    """Base Event class."""
+
+    _attr_entity_registry_enabled_default = False
+
+    @callback
+    def _async_handle_event(self, event: str) -> None:
+        """Handle event."""
+        self._trigger_event(event)
+        self.async_write_ha_state()
```

Comment:
Thank you for the review!
The HACS version of this integration has several types of Event entities, so a base class makes sense there. This PR adds the first Event type, so that is why there is groundwork of several types of Event entities for subsequent PRs. I would prefer to leave it as is, but I can "clean" the code for this PR if that is preferred.

Reply:
we usually like to only add stuff that is actually used, so yes please clean up